snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/normalform/NormalFormGenerator.java

```diff
@@ -531,11 +533,25 @@ private Iterable<StatementFragment> relationshipsFromUnionGroup(final NormalForm
 				});
 	}
 
-	@Deprecated
 	private Iterable<ConcreteDomainFragment> membersFromGroupSet(final NormalFormGroupSet targetGroupSet) {
-		// We will consume CD member fragments, but no longer suggest to create new ones.
-		return List.of();
+		// We will consume CD member fragments, but no longer suggest to create new ones, unless explicitly requested via inferConcreteDomainRefsetMembers option
+		return inferConcreteDomainRefsetMembers ? FluentIterable.from(targetGroupSet).transformAndConcat(this::membersFromGroup) : List.of();
 	}
+	
+	private Iterable<ConcreteDomainFragment> membersFromGroup(final NormalFormGroup group) {
+		return FluentIterable
+				.from(group.getUnionGroups())
+				.transformAndConcat(unionGroup -> membersFromUnionGroup(unionGroup, 
+					group.getGroupNumber(), 
+					unionGroup.getUnionGroupNumber()));
+	}
+
+	private Iterable<ConcreteDomainFragment> membersFromUnionGroup(final NormalFormUnionGroup unionGroup, final int groupNumber, final int unionGroupNumber) {
```

Comment:
Since CD members have no field to store a union group number in, we can either leave this off and/or check whether somehow a refset member ended up in a non-zero union group during the collection process.